packages/react-scripts/fixtures/kitchensink/src/features/env/NodePath.js

```diff
@@ -5,12 +5,17 @@ export default class extends React.Component {
   constructor(props) {
     super(props);
 
+    this.done = () => {};
+    this.props.setCallWhenDone && this.props.setCallWhenDone((done) => {
```

Comment:
This looks pretty complicated. Could you explain why just passing `onDone` as a prop, and then calling it in `setState` callback not enough?